pkg/network/dns_parser.go

```diff
@@ -160,47 +159,38 @@ func (p *dnsParser) parseAnswerInto(
 		return nil
 	}
 
-	var alias []byte
-	domainQueried := question.Name
+	t.dns = string(question.Name)
 	pktInfo.queryType = QueryType(question.Type)
-
-	// Retrieve the CNAME record, if available.
-	alias = p.extractCNAME(domainQueried, dns.Answers)
-	if alias == nil {
-		alias = p.extractCNAME(domainQueried, dns.Additionals)
-	}
-
-	// Get IPs
-	p.extractIPsInto(alias, domainQueried, dns.Answers, t)
-	p.extractIPsInto(alias, domainQueried, dns.Additionals, t)
-	t.dns = strings.ToLower(string(domainQueried))
+	alias := p.extractCNAME(t.dns, dns.Answers)
+	p.extractIPsInto(alias, dns.Answers, t)
+	t.dns = strings.ToLower(t.dns)
 
 	pktInfo.pktType = SuccessfulResponse
 	return nil
 }
 
-func (*dnsParser) extractCNAME(domainQueried []byte, records []layers.DNSResourceRecord) []byte {
+func (*dnsParser) extractCNAME(domainQueried string, records []layers.DNSResourceRecord) string {
+	alias := domainQueried
 	for _, record := range records {
-		if record.Type == layers.DNSTypeCNAME && record.Class == layers.DNSClassIN &&
-			bytes.Equal(domainQueried, record.Name) {
-			return record.CNAME
+		if record.Class != layers.DNSClassIN {
+			continue
+		}
+		if record.Type == layers.DNSTypeCNAME && alias == string(record.Name) {
+			alias = string(record.CNAME)
 		}
 	}
-
-	return nil
+	return alias
 }
 
-func (*dnsParser) extractIPsInto(alias, domainQueried []byte, records []layers.DNSResourceRecord, t *translation) {
+func (*dnsParser) extractIPsInto(alias string, records []layers.DNSResourceRecord, t *translation) {
```

Comment:
is there a good reason for replacing the `[]byte` for `string` everywhere? it looks like we're allocating a bunch of them now for no reason

Reply:
We eventually want `string`, and `bytes.Equal` casts to `string` anyways:
https://github.com/golang/go/blob/fd4b587da3f9a2bde193a5b9fd2ba96667e08f2d/src/bytes/bytes.go#L18-L21

Reply:
humm I didn't know about that, but according to the comment *that* might be optimized somehow by the compiler, whereas I'm almost sure that we're allocating. would you mind adding a benchmark for that? I feel we just need to allocate *once* before returning the function

Reply:
I switched back to using `[]byte`.